packages/@azure/keyvault/tests/secrets.test.ts

```diff
@@ -0,0 +1,126 @@
+import * as assert from "assert";
+import chai from "chai"
+const expect = chai.expect;
+import { getKeyvaultName, getCredentialWithServicePrincipalSecret, getUniqueName } from "./utils/utils.common";
+import { SecretsClient, Pipeline } from "../lib/secretsClient";
+import { signingPolicy, exponentialRetryPolicy, deserializationPolicy, ServiceClientCredentials, RestError } from '@azure/ms-rest-js';
+import { stringify } from 'querystring';
+
+describe("Secret client", () => {
+  let credential: ServiceClientCredentials;
+  let keyVaultName: string;
+  let keyVaultUrl: string;
+  let client: SecretsClient;
+
+  before(async () => {
+    credential = await getCredentialWithServicePrincipalSecret()
+    keyVaultName = getKeyvaultName();
+    keyVaultUrl = `https://${keyVaultName}.vault.azure.net`;
+    client = new SecretsClient(keyVaultUrl, credential);
+  });
+
+  it("can add a secret", async () => {
+    const secretName = getUniqueName("secret");
+    const secretValue = getUniqueName("value");
+    after(async () => {
+      await client.deleteSecret(secretName);
+    });
+
+    const result = await client.setSecret(secretName, secretValue);
+
+    assert.equal(result.name, secretName, "Unexpected secret name in result from setSecret().");
+    assert.equal(result.value, secretValue, "Unexpected secret value in result from setSecret().");
+  });
+
+  it("can retrieve the latest version of a secret value", async () => {
+    const secretName = getUniqueName("secret");
+    const secretValue = getUniqueName("value");
+    after(async () => {
+      await client.deleteSecret(secretName);
+    });
+    await client.setSecret(secretName, secretValue);
```

Comment:
shouldn't we set the secrets twice and then ensures that the value getSecret gives is the latest one?

Reply:
Agree that adding two secret would make it more explicit. The focus of this test is the optional parameter (and the only secret value that is just set also happened to be the latest 😄)